modules/jdbc/src/main/java/org/apache/ignite/internal/jdbc/JdbcStatement.java

```diff
@@ -90,6 +92,8 @@ public class JdbcStatement implements Statement {
     /** Current result index. */
     private int curRes;
 
+    private volatile @Nullable Long lastCorrelationToken;
```

Comment:
why not a just simple type 'long' and special value instead of null?

Reply:
first, I believe this is semantically more correct (null means there is no last value). Second, given that token generator wraps around, I don't want to deal with collision with this special value in generator (even though the chance to stumble upon collision is close to zero)